juspay/hyperswitch
# refactor(core): move update trackers after build request

## Type of Change



## Description

Update trackers has been moved before callling the connector. This is to make sure that if there is any validation error, then the payment status will not be updated. 
For other flows where connector does not need to be called ( except `session_token` flow, there is an issue to move it outside payment core #888), it will be called as normal.


## Motivation and Context

Fixes #1471 

## How did you test it?

- Create a payment with zen, without email - Payment status is `requires_payment_method`.
- Confirm the payment - error is Missing required param: email
- Retrieve the payment - status will be `requires_payment_method`.

## Checklist
